FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Common/Extensions/IServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,188 @@
+                throw new ConfigurationErrorException("Failed to parse job configuration", ex);
+            }
+
+            if (string.IsNullOrEmpty(jobConfiguration.ContainerName))
+            {
+                throw new ConfigurationErrorException($"Target azure container name can not be empty.");
+            }
+
+            FilterConfiguration filterConfiguration;
+            try
+            {
+                filterConfiguration = services
+                    .BuildServiceProvider()
+                    .GetRequiredService<IOptions<FilterConfiguration>>()
+                    .Value;
+            }
+            catch (Exception ex)
+            {
+                throw new ConfigurationErrorException("Failed to parse filter configuration", ex);
+            }
+
+            ValidateFilterConfiguration(filterConfiguration);
+
+            var storeConfiguration = services
+                .BuildServiceProvider()
+                .GetRequiredService<IOptions<DataLakeStoreConfiguration>>()
+                .Value;
+
+            if (string.IsNullOrEmpty(storeConfiguration.StorageUrl))
+            {
+                throw new ConfigurationErrorException($"Target azure storage url can not be empty.");
+            }
+
+            SchemaConfiguration schemaConfiguration;
+            try
+            {
+                schemaConfiguration = services
+                    .BuildServiceProvider()
+                    .GetRequiredService<IOptions<SchemaConfiguration>>()
+                    .Value;
+            }
+            catch (Exception ex)
+            {
+                throw new ConfigurationErrorException("Failed to parse schema configuration", ex);
+            }
+
+            if (schemaConfiguration.EnableCustomizedSchema)
+            {
+                if (string.IsNullOrWhiteSpace(schemaConfiguration.SchemaImageReference))
+                {
+                    throw new ConfigurationErrorException($"Customized schema image reference can not be empty when customized schema is enable.");
+                }
+                else
+                {
+                    ValidateImageReference(schemaConfiguration.SchemaImageReference);
+                }
+            }
+            else
+            {
+                if (!string.IsNullOrWhiteSpace(schemaConfiguration.SchemaImageReference))
+                {
+                    throw new ConfigurationErrorException($"Found the schema image reference but customized schema is disable.");
+                }
+            }
+        }
+
+        /// <summary>
+        /// Validate the image reference.
+        /// Image reference pattern: <registry>/<name>@<digest> or <registry>/<name>:<tag>
+        /// E.g. testacr.azurecr.io/templatetest@sha256:412ea84f1bb1a9d98345efb7b427ba89616ec29ac332d543eff9a2161ca12a58
+        /// </summary>
+        /// <param name="imageReference">Image reference.</param>
+        private static void ValidateImageReference(string imageReference)
```

Comment:
Can we add some tests for this acr reference validate interface? Maybe we can make it a public standalone method for testing purpose.

Reply:
Updated.